user/tests/wiring/ble/heartrate_central/application.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ ******************************************************************************
+  Copyright (c) 2015 Particle Industries, Inc.  All rights reserved.
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation, either
+  version 3 of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this program; if not, see <http://www.gnu.org/licenses/>.
+  ******************************************************************************
+ */
+
+/* Includes ------------------------------------------------------------------*/
+#include "application.h"
+
+BLEScanResult    results[10];
+
+BLEPeerAttribute heartrate;
+BLEPeerDevice    peer;
+
+void heartrateUpdated(uint8_t* data, size_t len) {
+
+}
+
+void setup() {
+}
+
+void loop() {
+    if (BLE.connected()) {
+        int newHr;
+        heartrate->getValue(&newHr);
+    }
+    else {
+        size_t count = BLE.scan(results, 10);
+
+        if (count > 0) {
+            for (uint8_t i = 0; i < count; i++) {
+                bool found = results[i].find(BLE_SIG_AD_TYPE_FLAGS);

```

Comment:
What does `find` do? What would the documentation say? Does it filter the results? Does it get the first result that matches? What are all the options that can be passed to it as a parameter?

Reply:
It checks whether the advertising contain the specific type of data. I have changed this API: https://github.com/particle-iot/device-os/pull/1728/files?#diff-75b7ed256204b5221da93d699fee0429R205